core/commands/UserResetPasswordCommand.php

```
@@ -90,13 +97,15 @@ function validate() {
 	protected function process() {
 		# If the password can be changed, we reset it, otherwise we unlock
 		# the account (i.e. reset failed login count)
-		$t_reset = auth_can_set_password( $this->user_id_reset );
-		if( $t_reset ) {
-			$t_result = user_reset_password( $this->user_id_reset );
+		if( auth_can_set_password( $this->user_id_reset ) ) {
+			$t_result = user_reset_password( $this->user_id_reset )
+				? self::RESULT_RESET
+				: self::RESULT_FAILURE; # Shouldn't we throw an exception in this case ?
```

Yes, we should throw an exception.